ipywidgets/widgets/widget_selectioncontainer.py

```diff
@@ -29,6 +30,11 @@ def _validated_index(self, proposal):
         else:
             raise TraitError('Invalid selection: index out of bounds')
 
+    @observe('children')
+    def _observe_children(self, change):
+        if self.selected_index is not None and len(change.new) < self.selected_index:
```

Comment:
This doesn't necessarily select the same child (if we delete a child from the front, now the selected container is not the same), which may be confusing.

Reply:
same as above